libmamba/include/mamba/specs/channel_spec.hpp

```diff
@@ -89,13 +89,16 @@ namespace mamba::specs
         ChannelSpec(std::string location, dynamic_platform_set filters, Type type);
 
         [[nodiscard]] auto type() const -> Type;
+        void set_type(Type t);
```

Comment:
I really don't like having setters for all members. This breaks the encapsulation and make it equivalent to have everything as public members.